UWP/UWPHelpers/StorageExtensions.cpp

```diff
@@ -0,0 +1,316 @@
+// UWP STORAGE MANAGER
+// Copyright (c) 2023 Bashar Astifan.
+// Email: bashar@astifan.online
+// Telegram: @basharastifan
+// GitHub: https://github.com/basharast/UWP2Win32
+
+// Functions:
+// replace(std::string& str, const std::string& from, const std::string& to)
+// replace2(const std::string str, const std::string& from, const std::string& to)
+// split(const std::string s, char seperator)
+// 
+// isChild(std::string parent, std::string child)
+// isParent(std::string parent, std::string child, std::string childName)
+// 
+// iequals(const std::string a, const std::string b) case-insenstive
+// equals(const std::string a, const std::string b) case-senstive
+// ends_with(std::string const& value, std::string const& ending)
+// starts_with(std::string str, std::string prefix)
+// 
+// convert(const std::string input)
+// convertToWString(const std::string input)
+// convert(Platform::String^ input)
+// convert(std::wstring input)
+// convert(const char* input)
+// convertToLPCWSTR(Platform::String^ input)
+// convertToLPCWSTR(std::string input)
+// convertToChar(Platform::String^ input)
+// 
+// tolower(std::string& input)
+// tolower(Platform::String^ &input)
+// toupper(std::string& input)
+// toupper(Platform::String^& input)
+// 
+// windowsPath(std::string& path)
+// windowsPath(Platform::String^ &path)
+// merge(std::string targetFullPath, std::string subFullPath)
+// 
+// findInList(std::list<T>& inputList, T& str)
+// isWriteMode(const char* mode)
+// getSubRoot(std::string parent, std::string child)
+
+#include "StorageExtensions.h"
+
+#pragma region Inernal Helpers
+std::string make_string(const std::wstring& wstring)
+{
+	if (wstring.empty())
+	{
+		return std::string();
+	}
+	size_t pos;
+	size_t begin = 0;
+	std::string ret;
+	size_t size;
+	pos = wstring.find(static_cast<wchar_t>(0), begin);
+	while (pos != std::wstring::npos && begin < wstring.length())
+	{
+		std::wstring segment = std::wstring(&wstring[begin], pos - begin);
+		size = WideCharToMultiByte(CP_UTF8, WC_ERR_INVALID_CHARS, &segment[0], (int)segment.size(), NULL, 0, NULL, NULL);
+		std::string converted = std::string(size, 0);
+		WideCharToMultiByte(CP_UTF8, WC_ERR_INVALID_CHARS, &segment[0], (int)segment.size(), &converted[0], (int)converted.size(), NULL, NULL);
+		ret.append(converted);
+		ret.append({ 0 });
+		begin = pos + 1;
+		pos = wstring.find(static_cast<wchar_t>(0), begin);
+	}
+	if (begin <= wstring.length())
+	{
+		std::wstring segment = std::wstring(&wstring[begin], wstring.length() - begin);
+		size = WideCharToMultiByte(CP_UTF8, WC_ERR_INVALID_CHARS, &segment[0], (int)segment.size(), NULL, 0, NULL, NULL);
+		std::string converted = std::string(size, 0);
+		WideCharToMultiByte(CP_UTF8, WC_ERR_INVALID_CHARS, &segment[0], (int)segment.size(), &converted[0], (int)converted.size(), NULL, NULL);
+		ret.append(converted);
+	}
+
+	return ret;
+}
+
+std::wstring make_wstring(const std::string& string)
```

Comment:
we have utilities for this kind of stuff already, in Common/Data/Encoding/Utf8.h . I guess it can still make sense to have it here if you want to make the StorageManager code reusable between apps.